content/en/docs/concepts/networking/function-entrypoints.md

```diff
@@ -57,32 +44,39 @@ curl -sv -HHOST:$FUNC_HOST http://$IP$FUNC_PATH
 
 ### Access functions by the external address
 
-#### Configure DNS
-You have two options to configure DNS:
-
-- Option 1: Real DNS
-
-    The real DNS configuration is based on the network topology of your environment.
-
-- Option 2: Configure the local hosts file
-
-    Get Function's HOSTNAME and Gateway's ip:
-    ```shell
-    export HOSTNAME="function-sample.default.ofn.io"
-    export IP=$(kubectl get node -l node-role.kubernetes.io/worker= -o=jsonpath='{.items[0].status.addresses[?(@.type=="InternalIP")].address}')
-    ```
-    
-    Append this line to /etc/hosts:
-    ```shell
-    echo $IP  $HOSTNAME | sudo tee -a /etc/hosts
-    ```
-
-#### Access Function
-Get `Function` external address by running following command:
+First, you should choose and configure a DNS based on your network environment:
+
+- Magic DNS
+
+  Get Kubernetes Gateway's ip address:
+  ```shell
+  export IP=$(kubectl get node -l "! node.kubernetes.io/exclude-from-external-load-balancers" -o=jsonpath='{.items[0].status.addresses[?(@.type=="InternalIP")].address}')
+  ```
+  Replace domain defined in OpenFunction Gateway with Magic DNS:
+  ```shell
+  export DOMAIN="$IP.sslip.io"
+  kubectl patch gateway.networking.openfunction.io/openfunction -n openfunction --type merge --patch '{"spec": {"domain": "'$DOMAIN'"}}'
+  ```
+  
+- Real DNS
+
+  If your networking environment produced an External IP address, then configure a wildcard A record for the domain:
+  ```
+  # Here example.com is the domain defined in OpenFunction Gateway
+  *.example.com == A <external-ip>
+  ```
+  If your networking environment produced a CNAME, then configure a CNAME record for the domain:
+  ```
+  # Here example.com is the domain defined in OpenFunction Gateway
+  *.example.com == A <external-name>
+  ```
+
+Then, you can get `Function` external address by running following command:
 ```shell
 export FUNC_EXTERNAL_ADDRESS=$(kubectl get function function-sample -o=jsonpath='{.status.addresses[?(@.type=="External")].value}')
 ```
-Access `Function` using curl directly:
+
```

Comment:
We need examples of function CR status containing `external address` for both magic DNS and real DNS method here to be more intuitive in yaml format.